src/ser/var.rs

```diff
@@ -46,18 +48,145 @@ where
         key: &K,
         value: &V,
     ) -> Result<()> {
-        // TODO: Is it possible to ensure our key is never a composite type?
-        // Anything which isn't a "primitive" would lead to malformed XML here...
-        write!(self.parent.writer, "<")?;
-        key.serialize(&mut *self.parent)?;
-        write!(self.parent.writer, ">")?;
+        let key = key.serialize(&mut MapKeySerializer)?;
```

Comment:
I made a very similar thing in #36 to detect whether something would be "wrapped" when serialized and [mentioned] we can use it to resolve this todo. I think it's a bit more general than your `MapKeySerializer`, so when #36 lands should we try to adapt this bit of code to use that? 

To make `WrapSafeDetector` more useful, it could probably return an enum which represents roughly which type of thing something  is (e.g. `Kind::String`, `Kind::Complex`, `Kind::Numeric`) with `is_wrapped()` being a helper method.

[mentioned]: https://github.com/RReverser/serde-xml-rs/pull/36#issuecomment-324070360